### PR TITLE
Fix: For MSSQL apply the same workaround for the varchar size as we use for Redshift

### LINKED ISSUE #2671: MSSQL tables with incorrect VARCHAR size
Looks like MSSQL is prone to the same issue as Redshift where tables get created with invalid size for some VARCHAR columns due to the fact that SQLMesh creates table using CTAS with `LIMIT 0`. Further investigation is required.

### PR DESCRIPTION
**Pull request by izeigerman**
Closes #2671

### CODE REVIEW

File: sqlmesh/core/engine_adapter/mixins.py [hunk 1]
```diff
@@ -246,3 +246,65 @@ def _truncate_table(self, table_name: TableName) -> None:
...
+
+        if (
+            statement.expression
+            and statement.expression.args.get("limit") is not None
+            and statement.expression.args["limit"].expression.this == "0"
+        ):
+            assert not isinstance(table_name_or_schema, exp.Schema)
+            # redshift and mssql have a bug where CTAS statements have non determistic types. if a limit
+            # is applied to a ctas statement, VARCHAR types default to 1 in some instances.
+            # this checks the explain plain from redshift and tries to detect when these optimizer
```
Comment (georgesittas) [inline sqlmesh/core/engine_adapter/mixins.py:280]:
```suggestion
            # this checks the explain plan from redshift and tries to detect when these optimizer
```

File: sqlmesh/core/engine_adapter/mixins.py [hunk 2]
```diff
@@ -246,3 +246,65 @@ def _truncate_table(self, table_name: TableName) -> None:
...
+            columns_to_types=columns_to_types,
+            table_description=table_description,
+            **kwargs,
+        )
+
+        if (
+            statement.expression
+            and statement.expression.args.get("limit") is not None
+            and statement.expression.args["limit"].expression.this == "0"
+        ):
```
Comment (georgesittas) [inline sqlmesh/core/engine_adapter/mixins.py:276]:
```suggestion
        limit = statement.expression.args.get("limit")
        if statement.expression and limit and limit.expression.this == "0":
```

**Review by georgesittas — APPROVED** [review]

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/engine_adapter/mixins.py (hunk 2 above)
Comment (izeigerman) [reply sqlmesh/core/engine_adapter/mixins.py:276]: `statement.expression` can be None too

Comment (georgesittas) [reply sqlmesh/core/engine_adapter/mixins.py:276]: yeah makes sense, missed that

**Pull request merged**